app.ts

```diff
@@ -0,0 +1,163 @@
+﻿/* Grid object class
+ *   this contains the entire grid object and methods to retrieve data from server 
+ *   and parse data to html strings for displaying in grid table on page
+ */
+class Grid {
+	element: HTMLElement;
+	nr_of_records: number;
+	nr_to_display: number;									
+	records_per_page: number;
+	display_from: number;
+	display_to: number;
+	gridHeadingString: string;
+
+	constructor(html_id: string) {							// html element id string 
+		this.element = document.getElementById(html_id)		// get DOM object ID. 
+		this.nr_of_records = 0;								// total nr of records on server
+		this.records_per_page = 30;							// display this many records per page
+		this.display_from = 0;								// display from this index
+		this.display_to = 0;								// display to this index
+		this.gridHeadingString = "";						// grid table heading 
+	}
+
+	setupEvents() {
+		let resizeTimer;
+		// 'debounce' window resize event to prevent multiple calls on each resize.
+		$(window).resize(() => {
+			let documentHeight = $(window).height(); //retrieve current document height
+			clearTimeout(resizeTimer);
+			resizeTimer = setTimeout(() => {
+				this.records_per_page = (Math.floor((documentHeight - 109) / 25)) - 2;
+				this.display_to = this.display_from + this.records_per_page;
+				this.getGridRecords();
+			}, 200);
+		});
+		// On-click event handler for PAGE DOWN button
+		$('#but_pagedown').click(() => {					// () => passes outside scope to inside function
+			this.page(this.records_per_page);
+		});
+		// On-click event handler for PAGE UP button
+		$('#but_pageup').click(() => {
+			this.page(-this.records_per_page);
+		});
+		// On-click event handler for GO button --> trigger .goto method
+		$('#but_go').click(() => {
+			this.goto();
+		});
+		// On-click event handler for HOME button
+		$('#but_home').click(() => {
+			$('#grid_show_from:text').val("0");
+			this.goto();
+		});
+		// Text input event handler for 'show from ID' input (prevents invalid input)
+		$('#grid_show_from:text').keypress( (e) => {
+			//if the letter is not a digit then don't type anything
+			if (e.which != 8 && e.which != 0 && (e.which < 48 || e.which > 57)) {
+				return false;
+			}
+		});
+	}
+
+/*  Goto: display records starting from a specific index value
+ *  a) read number from html input box, adjust .display_from and -to. 
+ *  b) check and readjust range if indices fall below 0 or above .nr_of_records
+ *  c) call .getGridRecords() method to fetch and display new record range
+ */
+	goto() {
+		let goto_id: string;
```

Comment:
Nitpicking: `goto_id` and `num`, Can assign their value directly.
`let goto_id = goto_id = $('#grid_show_from:text').val();`
`let num = parseInt(goto_id);`